server/APIrouter.js

```diff
@@ -0,0 +1,179 @@
+    const foundUser = getUser(req.params.username);
+    if (!(foundUser instanceof Error)) {
+        //User is found. Send the user data.
+        res.status(200).send(foundUser);
+    }
+    //User not found error. Send 404 response.
+    res.status(404).send(foundUser.message);
+});
+
+APIrouter.get('/getUserById/:id', (req, res, next) => {
+    const foundUser = getUserById(req.params.id);
+    if (!(foundUser instanceof Error)) {
+        res.status(200).send(foundUser);
+    }
+    res.status(404).send(foundUser.message);
+});
+
+//We are using a PUT request instead of a GET request so that the user can
+//send the password in the request body for secure (HTTPS) encyrption instead of through the URL.
+//This way the password is not exposed in plain text.
+APIrouter.put('/getUserByPasswd/:username', (req, res, next) => {
+    const foundUser = getUserByPasswd(req.params.username, req.body.password);
+    if (!(foundUser instanceof Error)) {
+        res.send(foundUser);
+    }
+    res.status(404).send(foundUser.message);
+});
+
+APIrouter.put('/loginUser/:username', (req, res, next) => {
+    //We need to wrap this in a self-executing async function, because the hashing library is also async.
+    (async () => {
+        const foundUser = await loginUser(req.params.username, req.body.password);
+        if (!(foundUser instanceof Error)) {
+            res.send(foundUser);
+        } else {
+            res.status(404).send(foundUser.message);
+        }
+    })();
+});
+
+//JSON Request Body Input:
+/*
+{"property": "your_property_here",
+ "value: "your_value_here"}
+
+For example if you want to change the age of a user to 5, the request body becomes:
+{"property": "age",
+ "value": 5}
+*/
+APIrouter.put('/updateUser/:username', (req, res, next) => {
+    const updatedUser = updateUser(req.params.username, req.body.property, req.body.value);
+    if (!(updatedUser instanceof Error)) {
+        res.send(foundUser);
+    }
+    res.status(404).send(updatedUser.message);
+});
+
+//JSON Request Body Input:
+//{"user:" {your_user_data_here}}
+APIrouter.post('/createUser/:username', (req, res, next) => {
+    //We need to wrap this in a self-executing async function, because the hashing library is also async.
+    (async () => {
+        /*  There is a bug here when: if there 2 create requests made one after another, both with the exact same data,
+            there will be an error and the server shuts down.
+            Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
+        */
+        const createdUser = await createUser(req.params.username, req.body.user);
+        if (!(createUser instanceof Error)) {
+            res.status(201).send(createdUser);
+        }
+        res.status(404).send(createdUser.message);
+    })();
+});
+
+APIrouter.delete('/deleteUser/:username', (req, res, next) => {
+    deleteUser(req.params.username);
+    res.status(204).send();
```

Comment:
I think it's better to explicitly inform the client that deleteUser succeeded/failed.